native/siwe_ex/src/lib.rs

```diff
@@ -75,17 +79,17 @@ fn message_to_parsed(m: Message) -> Parsed {
 }
 
 #[rustler::nif]
-fn from_str(message: String) -> NifResult<Parsed> {
+fn parse(message: String) -> Result<Parsed, String> {
     Ok(message_to_parsed(
-        Message::from_str(&message).map_err(|_e| Error::BadArg)?,
+        Message::from_str(&message).map_err(|e| format!("Failed to parse: {}", e))?,
     ))
 }
 
 #[rustler::nif]
-fn to_str(message: Parsed) -> NifResult<String> {
+fn to_str(message: Parsed) -> Result<String, String> {
     Ok(message
         .to_eip4361_message()
-        .map_err(|_e| Error::BadArg)?
+        .map_err(|e| format!("Faile to marshal to string: {}", e))?
```

Comment:
Oh sorry just noticed a small typo here too: `Faile` -> `Failed`

Reply:
Thank you! Wouldn't have caught that.